pandas/io/json/_normalize.py

```diff
@@ -74,10 +112,39 @@ def nested_to_record(
      'nested.e.c': 1,
      'nested.e.d': 2}
     """
+
+    def is_key_match(key, use_keys):
+        if callable(use_keys):
+            return use_keys(key)
+
+        if use_keys is None:
+            return lambda x: True
+
+        if isinstance(use_keys, str):
+            use_keys = [use_keys]
+
+        if is_list_like(use_keys):
+            return any(
+                key.split(".")[-len(i.split(".")) :] == i.split(".") for i in use_keys
```

Comment:
Can you break this up into a few lines instead of just one generator expression? I think would be more readable that way